dbt/adapters/snowflake/connections.py

```diff
@@ -455,6 +471,24 @@ def add_query(self, sql, auto_begin=True, bindings=None, abridge_sql_log=False):
             if without_comments == "":
                 continue
 
+            # Even though we turn off transactions by default for Snowflake,
+            # the user/macro has passed them *explicitly*, probably to wrap a DML statement
+            # Let their wish be granted!
+            # This also has the effect of ignoring "commit" in the RunResult for this model
+            # https://github.com/dbt-labs/dbt-snowflake/issues/147
+            if individual_query.lower() == "begin;":
```

Comment:
Feels a bit jank, but actually:
- These are generally being added by our own internal macro, `snowflake_dml_explicit_transaction`, and we know exactly what it's adding
- We've already stripped whitespace above
- We use the Snowflake connector's official [`split_statements`](https://github.com/snowflakedb/snowflake-connector-python/blob/37443fb4cd0c548df0efe62199d2b52b324abe6e/src/snowflake/connector/util_text.py#L32), which seems to return statements with the `;` included.
- I'm actually not sure why we're not also using their official `remove_comments` argument to `split_statements`, rather than [our own janky logic](https://github.com/dbt-labs/dbt-snowflake/blob/ae31719cb0c575460e647071e464f8e7344749ec/dbt/adapters/snowflake/connections.py#L446-L453) here? Most recent work there was https://github.com/dbt-labs/dbt-core/pull/2974
- Worst case, we don't catch these as "real" `begin` + `commit`, just send them along as normal queries, and store their results accordingly